File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/MaskPositionFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.FloatFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;
import ru.ioleksiv.telegram.bot.api.model.annotation.CustomValidator;
import ru.ioleksiv.telegram.bot.api.model.annotation.stub.StubMaskPositionValidator;
import ru.ioleksiv.telegram.bot.api.model.objects.std.sticker.MaskPosition;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface MaskPositionFilter {

    AnnotationState status() default AnnotationState.ON;

    String[] validator() default {};

    MaskPosition.Type point() default MaskPosition.Type.ALL;

    FloatFilter xShift() default @FloatFilter(status = AnnotationState.OFF);

    FloatFilter yShift() default @FloatFilter(status = AnnotationState.OFF);

    FloatFilter scale() default @FloatFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/MaskPositionFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.FloatFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState
import io.github.oleksivio.telegram.bot.api.model.objects.std.sticker.MaskPosition

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class MaskPositionFilter(
        val status: AnnotationState = AnnotationState.ON,
        val validator: Array<String> = [],
        val point: MaskPosition.Type = MaskPosition.Type.UNKNOWN,
        val xShift: FloatFilter = FloatFilter(status = AnnotationState.OFF),
        val yShift: FloatFilter = FloatFilter(status = AnnotationState.OFF),
        val scale: FloatFilter = FloatFilter(status = AnnotationState.OFF)
)
